app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepositoryImpl.kt

```diff
@@ -0,0 +1,308 @@
+import com.realityexpander.tasky.agenda_feature.data.repositories.attendeeRepository.IAttendeeRepository
+import com.realityexpander.tasky.agenda_feature.data.repositories.syncRepository.ISyncRepository
+import com.realityexpander.tasky.agenda_feature.data.repositories.syncRepository.local.AgendaItemTypeForSync
+import com.realityexpander.tasky.agenda_feature.data.repositories.syncRepository.local.ModificationTypeForSync
+import com.realityexpander.tasky.agenda_feature.domain.*
+import com.realityexpander.tasky.auth_feature.domain.IAuthRepository
+import com.realityexpander.tasky.core.util.Email
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
+import java.time.ZonedDateTime
+import javax.inject.Inject
+
+class AgendaRepositoryImpl @Inject constructor(
+    private val agendaApi: IAgendaApi,
+    private val eventRepository: IEventRepository,
+    private val attendeeRepository: IAttendeeRepository,
+    private val reminderRepository: IReminderRepository,
+    private val taskRepository: ITaskRepository,
+    private val syncRepository: ISyncRepository,
+    private val authRepository: IAuthRepository
+) : IAgendaRepository {
+
+    ///////////////////////////////////////////////
+    // • AGENDA
+
+    override suspend fun getAgendaForDay(dateTime: ZonedDateTime): List<AgendaItem> {
+        val events = eventRepository.getEventsForDay(dateTime)
+        val tasks = taskRepository.getTasksForDay(dateTime)
+        val reminders = reminderRepository.getRemindersForDay(dateTime)
+        return events + tasks + reminders
+    }
+
+    override suspend fun getAgendaForDayFromRemote(dateTime: ZonedDateTime): Result<List<AgendaItem>> {
+        return try {
+            val result = agendaApi.getAgenda(dateTime)
+            if (result.isSuccess) {
+                val agendaDayDTO = result.getOrNull()
+                if (agendaDayDTO != null) {
+                    val events = agendaDayDTO.events.map { it.toDomain() }
+                    val tasks = agendaDayDTO.tasks.map { it.toDomain() }
+                    val reminders = agendaDayDTO.reminders.map { it.toDomain() }
+                    Result.success(events + tasks + reminders)
+                } else {
+                    Result.failure(Exception("Response body is null"))
+                }
+            } else {
+                Result.failure(Exception("Error getting agenda"))
+            }
+        } catch (e: Exception) {
+            Result.failure(Exception("Error getting agenda"))
+        }
+    }
+
+    override suspend fun updateLocalAgendaForDayFromRemote(dateTime: ZonedDateTime) {
+        CoroutineScope(Dispatchers.IO).launch {
```

Comment:
The scope should be removed here, in most cases you just want to expose suspend functions and execute them in `viewModelScope`, since otherwise you have no control over the execution of this function (for example if you want to cancel it or know when it finished)

Reply:
ok i will refactor